src/video_core/renderer_vulkan/vk_pipeline_cache.cpp

```diff
@@ -33,49 +33,52 @@ constexpr static std::array DescriptorHeapSizes = {
     vk::DescriptorPoolSize{vk::DescriptorType::eSampler, 1024},
 };
 
-void GatherVertexOutputs(Shader::VertexRuntimeInfo& info,
-                         const AmdGpu::Liverpool::VsOutputControl& ctl) {
-    const auto add_output = [&](VsOutput x, VsOutput y, VsOutput z, VsOutput w) {
-        if (x != VsOutput::None || y != VsOutput::None || z != VsOutput::None ||
-            w != VsOutput::None) {
-            info.outputs[info.num_outputs++] = Shader::VsOutputMap{x, y, z, w};
-        }
-    };
+void GatherMiscOutputs(Shader::RuntimeInfo& info, const AmdGpu::Liverpool::VsOutputControl& ctl) {
     // VS_OUT_MISC_VEC
-    add_output(ctl.use_vtx_point_size ? VsOutput::PointSprite : VsOutput::None,
-               ctl.use_vtx_edge_flag
-                   ? VsOutput::EdgeFlag
-                   : (ctl.use_vtx_gs_cut_flag ? VsOutput::GsCutFlag : VsOutput::None),
-               ctl.use_vtx_kill_flag
-                   ? VsOutput::KillFlag
-                   : (ctl.use_vtx_render_target_idx ? VsOutput::GsMrtIndex : VsOutput::None),
-               ctl.use_vtx_viewport_idx ? VsOutput::GsVpIndex : VsOutput::None);
+    auto& outputs = info.GetOutputs();
+    if (ctl.vs_out_misc_enable) {
+        auto& misc_vec = outputs[0];
```

Comment:
Using static index 0/1/2 here is incorrect. If VS_OUT_MISC_VEC for example is disabled, then the next one VS_OUT_CCDIST0 will occupy the first output. Define a local num_outputs and index with num_outputs++ here and other branches